088-rebalance-progress-status.md

```diff
@@ -0,0 +1,390 @@
+- **completedByteMovementPercentage**: The percentage of the byte movement of the partition rebalance that is completed as a rounded down integer value in the range [0-100].
+- **executorState**: The “non-verbose” JSON payload from the [`/kafkacruisecontrol/state?substates=executor`](https://github.com/linkedin/cruise-control/wiki/REST-APIs#query-the-state-of-cruise-control) endpoint, providing details about the executor's current status, including partition movement progress, concurrency limits, and total data to move.
+
+Since the progress information is constant, we can safely add it to the existing `ConfigMap` maintained for and tied to the `KafkaRebalance` resource.
+This keeps `KafkaRebalance` information organized in one place, simplifies the proposal implementation, and has insignificant impact on the storage of the `ConfigMap`.
+
+The enhanced `ConfigMap` of an inter-broker partition rebalance will look like the following:
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: my-rebalance
+  …
+data:
+  estimatedTimeToCompletionInMinutes: 5 [1]
+  completedByteMovementPercentage: 80 [2]
+  executorState: [3]
+  {
+   "abortingPartitions":0,
+   "averageConcurrentPartitionMovementsPerBroker":5,
+   "finishedDataMovement":0,
+   "maximumConcurrentPartitionMovementsPerBroker":5,
+   "minimumConcurrentPartitionMovementsPerBroker":5,
+   "numFinishedPartitionMovements":0,
+   "numInProgressPartitionMovements":0,
+   "numPendingPartitionMovements":20,
+   "numTotalPartitionMovements":20,
+   "state":"INTER_BROKER_REPLICA_MOVEMENT_TASK_IN_PROGRESS",
+   "totalDataToMove":0,
+   "triggeredSelfHealingTaskId":"",
+   "triggeredTaskReason":"No reason provided (Client: 172.17.0.1, Date: 2024-11-15T19:41:27Z)",
+   "triggeredUserTaskId":"0230d401-6a36-430e-9858-fac8f2edde93"
+  }
+  brokerLoad.json: {...} [4]
+```
+[1] The estimated time it will take in minutes for the rebalance to complete based on the average rate of data transfer.
+
+[2] The percentage of byte movement of the partition balance that is complete as a rounded down integer in the range [0-100].
+
+[3] The “non-verbose” JSON payload from [/kafkacruisecontrol/state?substates=executor](https://github.com/linkedin/cruise-control/wiki/REST-APIs#query-the-state-of-cruise-control) endpoint.
+
+[4] The broker load from the optimization proposal as a JSON string that already maintained in the `ConfigMap`.
+
+### Field values per `KafkaRebalance` State
+
+We will provide the progress information for the `KafkaRebalance` states where it is relevant:
+
+
+| State           | `estimatedTimeToCompletionInMinutes`                                   | `completedByteMovementPercentage`                                                                                                    | `executorState`                                                                                                                                                                                      |
```

Comment:
I would use "State/Field" in the first cell to make it clear, rows are states and columns are fields.